### PR TITLE
Improve machinery for testing consistency in presence of range movements

### PR DESCRIPTION
**Pull request by ifesdjeen**
Patch by Alex Petrov; reviewed by TBD for CASSANDRA-15935

### CODE REVIEW

File: src/main/java/org/apache/cassandra/distributed/api/IInstance.java [hunk 1]
```diff
@@ -23,9 +23,15 @@
 import java.util.UUID;
 import java.util.concurrent.Future;
 
+import org.apache.cassandra.distributed.shared.VersionedApplicationState;
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/IInstance.java]: this isn't used, so looks like dead code?

File: src/main/java/org/apache/cassandra/distributed/shared/VersionedApplicationState.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.shared;
+
+import java.io.Serializable;
+
+public class VersionedApplicationState implements Serializable
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/shared/VersionedApplicationState.java:23]: looks like dead code, I would remove.

File: src/main/java/org/apache/cassandra/distributed/api/IInstance.java [hunk 3]
```diff
@@ -23,9 +23,15 @@
...
+
 // The cross-version API requires that an Instance has a constructor signature of (IInstanceConfig, ClassLoader)
 public interface IInstance extends IIsolatedExecutor
 {
+    enum StartupOption
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/api/IInstance.java]: If we leave this in, I would recommend the following

```
interface StartupOption {}
enum StandardStartupOptions { SKIP_RING_JOIN }
```

this enables us to make forward progress without requiring a new release of the interface every time we need a new option.

**Comment by ifesdjeen** [comment]
As discussed with @dcapwell, since we don't strictly need dtest changes to bring in support of range movements, leaving this for the future if/when we decide to support multi-version gossip and ring movement tests.

**Pull request closed without merging**